### ISSUE TITLE
Локализация документации

### ISSUE THREAD

**Issue by arhangelsoft** [zenden2k/image-uploader#132]
Надо ли и как её делать? В смысле оригинал на английском, и копия на русском.

**Comment by zenden2k** [comment]
Создал папку ru_RU и en_US.
https://github.com/zenden2k/image-uploader/tree/1.3.2/Docs
API только на английском.
Если будете что-то делать (кроме добавления хостингов), работайте над этой веткой.

**Comment by arhangelsoft** [comment]
>API только на английском.

Бывает так что человек учил например немецкий и французкий. А MSDN на русском та ещё хохма)

**Comment by zenden2k** [comment]
Поддерживать разные языковые версии в актуальном состоянии будет тяжело, я и с одной-о версией не успеваю, да и не нужно это. 
Программисту без английского никуда.

MSDN вообще хохма, там половина ссылок битые, а навигация вообще ни в какие ворота не лезет.

**Comment by zenden2k** [comment]
Ужас, сколько фич придумали, когда я это всё разгребу?

**Comment by arhangelsoft** [comment]
В C++ я начинающий(ну по крайней мере наловчился переводить C++ код на паскаль), еле разобрался как проект собрать в MSVS :( 

Зачем такой геморрой?

Можно на Angular.JS или даже jQuery + Bootstrap написать одностраничное приложение, которое ajax-ом будет грузить и папки с локалями версию для того или иного языка.

./index.itml - шапка+меню и подвал
./main/ru/... - тушки для контента index.html
./main/en/...- тушки для контента index.html
./api/ru/...
./api/en/...

В топку HTML, можно сразу на wiki-движек.

**Comment by arhangelsoft** [comment]
Собственно сам двиг wiki http://www.mediawiki.org/

**Comment by arhangelsoft** [comment]
могу его у себя разместить, не сложно)

**Comment by zenden2k** [comment]
Насчет шапки шапки и футера согласен, их надо вынести в отдельный файл, тут уже возникают вопросы кроссбраузерности, еще их мне не хватало, главное, чтобы я при желании мог дать ссылку на нужное место нужного документа.

>Или вы хотите актуализировать документацию автоматом из исходников?
Да.

>Такие доки хуже чем MSDN на английском через google-translate на русском.
Чем оно хуже того, что есть на данный момент?

**Comment by arhangelsoft** [comment]
>Насчет шапки шапки и футера согласен, их надо вынести в отдельный файл, но насчет одностраничного приложения, я не уверен, тут уже возникают вопросы кроссбраузерности, еще их мне не хватало, главное, чтобы я при желании мог дать ссылку на нужное место нужного документа.

jQuery кросс-браузерный до мозга костей, bootstrap вообще для кросс-браузерности и создавался( набор HTML элементов для шустрой верстки крутых штук одними лишь каскадными стилями). 

ссылку тоже можно организовать. Angular.JS так и работает.

>Чем оно хуже того, что есть на данный момент?

Документация создана машиной. Это видно и это чувствуется при чтении. Машина не пишет примеры кода и понятные объяснения.

По крайней мере Doc-o-Matic такой. нашел тут приемлемый вам генератор: http://www.doxys.dk/doxys_homepage/index.html

легенький, простенький.

**Comment by zenden2k** [comment]
>Документация создана машиной. Это видно и это чувствуется при чтении. Машина не пишет примеры кода и понятные объяснения.

ЭЭЭ так описание методов я сам буду писать. и примеры кода doxygen тоже поддерживает.

**Comment by zenden2k** [comment]
>ссылку тоже можно организовать. Angular.JS так и работает.
ну тогда дерзайте)

**Comment by arhangelsoft** [comment]
Так doxygen или руками?)

**Comment by zenden2k** [comment]
api пока не надо локализовывать

**Comment by arhangelsoft** [comment]
И все же лучше сделать на wikimedia, меньше гемороя, для локализации уже все есть. Можно перед релизом обновить доки и экпортировать в тот же pdf или html и отдать инсталятору.

**Comment by zenden2k** [comment]
Наоборот, больше геммороя. 
Зачем делать вики ради 3-х страничек и двух человек? Документация по API будет всё-равно генерироваться автоматически.

**Comment by zenden2k** [comment]
Вспомнил, что на гугл коде был вики.
Но при переезде на гитхаб wiki разметка превратилась в markdown.
https://github.com/zenden2k/image-uploader/blob/wiki/ProjectHome.md

**Comment by zenden2k** [comment]
На гитхабе тоже есть вики
https://github.com/zenden2k/image-uploader/wiki/Image-Uploader-Wiki

**Comment by arhangelsoft** [comment]
>Я хочу, чтобы документация была в репозитории.

хранить pdf'ки на гитхабе)

**Comment by zenden2k** [comment]
Вот что получилось с API
http://zenden2k.github.io/image-uploader/api/html/annotated.html

**Comment by arhangelsoft** [comment]
>И Doxygen кстати не поддерживает локализацию.

поддерживает: http://www.stack.nl/~dimitri/doxygen/manual/langhowto.html

**Comment by arhangelsoft** [comment]
плохо то, что генерится из исходника и если я начну переводить, это теоретически сгенерирует кучу лулзов.

**Comment by zenden2k** [comment]
>поддерживает: 

Это локализация самого doxygen, т.е. локализация шаблонов, а не локализация документации

**Comment by zenden2k** [comment]
Ну, если найдете средство, совместимое с doxygen разметкой и поддерживающее локализацию, тогда можно будет переводить. Или написать что-то своё. 
Но у меня это делать нет ни времени, ни желания.
И так очень много еще работы. Многопоточную загрузку всё никак не могу закончить
+ начал делать кроссплатформенную версию https://github.com/zenden2k/image-uploader/tree/1.3.2/Source/ImageUploaderQt

**Comment by arhangelsoft** [comment]
все же, что на счёт этого? http://www.di-mare.com/adolfo/p/DoxEsEn%28en%29-LACCEI-09.pdf

**Comment by zenden2k** [comment]
неееееее

**Comment by arhangelsoft** [comment]
русский и английский в самый раз)

**Comment by arhangelsoft** [comment]
какая версия QT?

**Comment by zenden2k** [comment]
5.4, а что?

**Comment by arhangelsoft** [comment]
мало ли, 4я)

**Comment by zenden2k** [comment]
В принципе, doxygen для каждого элемента документации генерирует уникальный id.
`https://zenden2k.github.io/image-uploader/api/html/class_script_a_p_i_1_1_process.html#abdb094a11ad94e0f1ec901c967ec2546`
Вот после символа '#', который
Можно написать парсер html (или чего-нибудь другого, потому что doxygen умеет много форматов, в .т.ч промежуточный xml формат), который после регенерации документации doxygen обновлял бы русский вариант, оставляя уже переведенные строки

**Comment by zenden2k** [comment]
Я бы может и написал этот парсер на php, но некогда.
Тем более у меня есть уже похожая задача по локализации, собираюсь перевести весь проект на [gettext](https://ru.wikipedia.org/wiki/Gettext).

**Comment by arhangelsoft** [comment]
это вы под QT уже собираетесь?

**Comment by zenden2k** [comment]
нет. Локализация основной программы + локализация консольной программы.
Правда, я не не знаю, как быть с Qt, потому что там своя локализация, похоже в Qt версии придется иметь 2 файла локализации.
